homeassistant/components/zha/helpers.py

```diff
@@ -509,6 +509,10 @@ def __init__(
         self._unsubs: list[Callable[[], None]] = []
         self._unsubs.append(self.gateway.on_all_events(self._handle_event_protocol))
         self._reload_task: asyncio.Task | None = None
+        self.hass.bus.async_listen(
+            er.EVENT_ENTITY_REGISTRY_UPDATED,
+            self._handle_entity_registry_updated,
+        )
```

Comment:
Should we wrap this function in `config_entry.async_on_unload()` to make sure its unloaded when the entry is unloaded?

Reply:
good call. Thx!